Parcelable[] cannot be cast to HybridFileParcelable[] at PasteHelper

This error surprised need so much that I don't remember what I had just done. I was going to save a photo in a Syncthing directory. I had just created that directory on a laptop. Possibly the Syncthing propagation timing was just right and confused Amaze.

Grizwolde Phantasia


## Exception
* __User Action:__ UI Error
* __Request:__ Application crash
* __Version:__ 3.5.2
* __OS:__ Linux samsung/d2que/d2q:10/QP1A.190711.020/N975U1UEU4DTH7:user/release-keys 10 - 29
* __Device:__ d2q
* __Model:__ SM-N975U1
* __Product:__ d2que
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.amaze.filemanager/com.amaze.filemanager.ui.activities.MainActivity}: java.lang.ClassCastException: android.os.Parcelable[] cannot be cast to com.amaze.filemanager.filesystem.HybridFileParcelable[]
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3654)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3806)
at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2267)
at android.os.Handler.dispatchMessage(Handler.java:107)
at android.os.Looper.loop(Looper.java:237)
at android.app.ActivityThread.main(ActivityThread.java:8167)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
Caused by: java.lang.ClassCastException: android.os.Parcelable[] cannot be cast to com.amaze.filemanager.filesystem.HybridFileParcelable[]
at com.amaze.filemanager.filesystem.PasteHelper.<init>(PasteHelper.java:73)
at com.amaze.filemanager.filesystem.PasteHelper.<init>(PasteHelper.java:52)
at com.amaze.filemanager.filesystem.PasteHelper$1.createFromParcel(PasteHelper.java:90)
at com.amaze.filemanager.filesystem.PasteHelper$1.createFromParcel(PasteHelper.java:88)
at android.os.Parcel.readParcelable(Parcel.java:2990)
at android.os.Parcel.readValue(Parcel.java:2883)
at android.os.Parcel.readArrayMapInternal(Parcel.java:3261)
at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:292)
at android.os.BaseBundle.unparcel(BaseBundle.java:236)
at android.os.BaseBundle.getBoolean(BaseBundle.java:899)
at android.app.Activity.restoreHasCurrentPermissionRequest(Activity.java:8287)
at android.app.Activity.performCreate(Activity.java:7959)
at android.app.Activity.performCreate(Activity.java:7952)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1307)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3629)
... 11 more

```
</details>
<hr>

--- a/app/src/main/java/com/amaze/filemanager/filesystem/PasteHelper.java
+++ b/app/src/main/java/com/amaze/filemanager/filesystem/PasteHelper.java
@@ -68,9 +68,7 @@ public final class PasteHelper implements Parcelable {
 
   private PasteHelper(Parcel in) {
     operation = in.readInt();
-    paths =
-        (HybridFileParcelable[])
-            in.readParcelableArray(HybridFileParcelable.class.getClassLoader());
+    paths = in.createTypedArray(HybridFileParcelable.CREATOR);
   }
 
   @Override
@@ -81,7 +79,7 @@ public final class PasteHelper implements Parcelable {
   @Override
   public void writeToParcel(Parcel dest, int flags) {
     dest.writeInt(operation);
-    dest.writeParcelableArray(paths, 0);
+    dest.writeTypedArray(paths, 0);
   }
 
   public static final Parcelable.Creator CREATOR =

